src/Microsoft.AspNet.Http.Features/Authentication/AuthenticateContext.cs

```diff
@@ -41,5 +41,12 @@ public virtual void NotAuthenticated()
         {
             Accepted = true;
         }
+
+        public bool IsRequestCompleted { get; private set; }
```

Comment:
I think IsRequestCompleted is redundant with Accepted (and possibly Response.HasStarted, depending on what you're trying to do).